src/param/param_type.rs

```diff
@@ -128,7 +122,42 @@ impl From<u16> for ParamType {
             49155 => ParamType::ErrClauseInd,
             49156 => ParamType::SetPriAddr,
             49157 => ParamType::SuccessInd,
-            _ => ParamType::Unknown,
+            49158 => ParamType::AdaptLayerInd,
+            unknown => ParamType::Unknown { param_type: unknown },
+        }
+    }
+}
+
+impl From<ParamType> for u16 {
+    fn from(v: ParamType) -> u16 {
+        match v {
```

Comment:
Isn't it better to keep the values on the enum variants and then just go `v as u16`?

Reply:
I think I did that because rust doesn't allow assigning specific numeric values to variants when there are fields on them

Reply:
fair enough!